### ISSUE TITLE
EOF error for non keep alive HTTP 1.1 requests

### ISSUE THREAD

**Issue by hellivan** [fortio/fortio#972]
**The version of the project you were using:** v1.66.2 (582d82a75a3172cc40eb0fe8e807c753b2a53770)

**Operating system you are using:** Arch Linux (6.10.5-arch1-1)

**The exact, minimal, steps needed to reproduce the issue:** Unfortunately I am not able to implement a minimal example that reliable reproduces the issue.

**Problem description:**
When i try to load test a specific application using the following command, i always get a `EOF` error resulting in a `Code  -1` statistic:
```
./fortio load -n 1 -c 1 --qps 0 -keepalive=false  https://somehost/api/infos
10:04:14.138 r1 [INF] scli.go:122> Starting, command="Φορτίο", version="dev  go1.23.0 amd64 linux", go-max-procs=8
Fortio dev running at 0 queries per second, 8->8 procs, for 1 calls: https://somehost/api/infos
10:04:14.139 r1 [INF] httprunner.go:121> Starting http test, run=0, url="https://somehost/api/infos", threads=1, qps="-1.0", warmup="parallel", conn-reuse=""
Starting at max qps with 1 thread(s) [gomax 8] for exactly 1 calls (1 per thread + 0)
10:04:14.153 r1 [ERR] http_client.go:1100> Read error, err="EOF", size=0, header_len=0, dest={"IP":"0.0.0.0","Port":443,"Zone":""}, url="https://somehost/api/infos", thread=0, run=0
10:04:14.154 r1 [INF] periodic.go:851> T000 ended after 12.940561ms : 1 calls. qps=77.27640246817738
Ended after 13.004645ms : 1 calls. qps=76.896
10:04:14.154 r1 [INF] periodic.go:581> Run ended, run=0, elapsed=13004645, calls=1, qps=76.8956015331445
Aggregated Function Time : count 1 avg 0.01293497 +/- 0 min 0.01293497 max 0.01293497 sum 0.01293497
# range, mid point, percentile, count
>= 0.012935 <= 0.012935 , 0.012935 , 100.00, 1
# target 50% 0.012935
# target 75% 0.012935
# target 90% 0.012935
# target 99% 0.012935
# target 99.9% 0.012935
Error cases : count 1 avg 0.01293497 +/- 0 min 0.01293497 max 0.01293497 sum 0.01293497
# range, mid point, percentile, count
>= 0.012935 <= 0.012935 , 0.012935 , 100.00, 1
# target 50% 0.012935
# target 75% 0.012935
# target 90% 0.012935
# target 99% 0.012935
# target 99.9% 0.012935
# Socket and IP used for each connection:
[0]   1 socket used, resolved to 0.0.0.0:443, connection timing : count 1 avg 0.007703781 +/- 0 min 0.007703781 max 0.007703781 sum 0.007703781
Connection time histogram (s) : count 1 avg 0.007703781 +/- 0 min 0.007703781 max 0.007703781 sum 0.007703781
# range, mid point, percentile, count
>= 0.00770378 <= 0.00770378 , 0.00770378 , 100.00, 1
# target 50% 0.00770378
# target 75% 0.00770378
# target 90% 0.00770378
# target 99% 0.00770378
# target 99.9% 0.00770378
Sockets used: 1 (for perfect keepalive, would be 1)
Uniform: false, Jitter: false, Catchup allowed: true
IP addresses distribution:
0.0.0.0:443: 1
Code  -1 : 1 (100.0 %)
Response Header Sizes : count 1 avg 0 +/- 0 min 0 max 0 sum 0
Response Body/Total Sizes : count 1 avg 0 +/- 0 min 0 max 0 sum 0
All done 1 calls (plus 0 warmup) 12.935 ms avg, 76.9 qps
```
Initially I debugged the client-server communication using wireshark, but did not find any abnormalities. That said, from the network perspective everything worked correctly, which means that all data was transferred completely from the server to the client and the and the connections (TLS as well as TCP) were closed normally.
I then started debugging fortio on the latest master branch (582d82a75a3172cc40eb0fe8e807c753b2a53770) and noticed that `fhttp/http_client.go:1074`  (`nI, err := conn.Read(c.buffer[c.size:])`) was actually returning the `EOF` error. When inspecting the content of `c.buffer` right after the `conn.Read` call I also noticed the the buffer already contained all the data sent by the server. `c.size` however was still 0 since this variable is increased at some later point (`fhttp/http_client.go:1104`). As far as I understand, `fhttp/http_client.go:1089` should handle the case where we get a `EOF` error as part of a normal end of stream connection termination. In this case however, the condition cannot evaluate to true since `c.size` is still `0`. As a result the request is interpreted as failed event though it was successful.
As far as I understand the issue is caused by the rather unusual behavior of the TLS socket's `Read` function which in this case reads the data completely and returns `EOF` error. Would it instead read the data and return `EOF` error on the next iteration call (just like in regular C code), then the current implementation would work correctly.

**Reproducing the problem:**
In my setup I was connecting to a nginx:1.27.1-alpine on the local lan that implemented TLS termination (TLS 1.2) and proxied the requests to a node.js application on the same lan, serving a static in-memory response. I noticed, that the problem could be reproduced more easily if the response from the node.js application was rather small (~413 bytes including headers) and the response was sent with very low latency.
NB: The same request using HTTP instead of HTTPS (still passing the nginx reverse proxy) never had these problems.

**Possible solution:**
IMO the following patch should fix the problem (at least it did in my case), however I'm not sure about the implications of such a change:
```
diff --git a/fhttp/http_client.go b/fhttp/http_client.go
index 21d5958..021a740 100644
--- a/fhttp/http_client.go
+++ b/fhttp/http_client.go
@@ -1086,7 +1086,7 @@ func (c *FastClient) readResponse(conn net.Conn, reusedSocket bool) {
                                        c.code = RetryOnce // special "retry once" code
                                        return
                                }
-                               if errors.Is(err, io.EOF) && c.size != 0 {
+                               if errors.Is(err, io.EOF) && (c.size > 0 || n > 0) {
                                        // handled below as possibly normal end of stream after we read something
                                        break
                                }
```

**Comment by ldemailly** [comment]
as a data point what output do you get with -stdclient ?

**Comment by ldemailly** [comment]
also are you using a binary distribution?
can you add the output of:

fortio buildinfo

**Comment by ldemailly** [comment]
ah! I see it dev  "go1.23.0 amd64 linux"

can you confirm it does or doesn't happen if you use the binary release or use go1.22.7 ?

**Comment by ldemailly** [comment]
it is a bug in the code because "Callers should always process the n > 0 bytes returned before considering the error err. Doing so correctly handles I/O errors that happen after reading some bytes and also both of the allowed EOF behaviors." in https://pkg.go.dev/io#Reader

yet the fix you have wouldn't be correct (it wouldn't process the data if you break)

will make a fix

**Comment by ldemailly** [comment]
@hellivan while I get what is likely happening, I am having a hard time reproducing it, even using stuff like

```
echo -n -e "HTTP/1.1 200 Ok\r\nContent-Length:0\r\n\r\n" | openssl s_server -accept 8777 -cert server.crt -key server.key
```
as the "server"

and
```
go1.23.0 run . load -loglevel debug -n 1 -c 1 -qps 0 -keepalive=false -cacert ./cert-tmp/ca.crt https://localhost:8777
```

Can you please rerun with `-loglevel debug`  if you are able to reproduce it

(I did get weird errors with std client like https://github.com/golang/go/issues/31259#issuecomment-2346981784 trying to compare with std client though)

**Comment by ldemailly** [comment]
also if you can reproduce it, can you confirm it's fixed with the branch from #974 : 
```
go run fortio.org/fortio@eof_handling  load -n 1 -c 1 --qps 0 -keepalive=false  https://...
```

**Comment by hellivan** [comment]
@ldemailly: Thanks for your investigation. Unfortunately I currently have no machine near me. However I will test your fix tomorrow morning as soon as possible and provide you with the requested information. For now I can tell you that I found this issue while testing our apis using the latest official fortio container image from dockerhub. In order to better analyze the problem I then compiled it from source (master branch).

**Comment by hellivan** [comment]
@ldemailly The container-image I used was
```
registry-1.docker.io/fortio/fortio:latest@sha256:0dec927c7f5030fc81f40b2665f1c244086b967b3ff04ce36ccea8dc359d68c8
```

* The build infos are the following:
```
docker run --rm registry-1.docker.io/fortio/fortio:latest@sha256:0dec927c7f5030fc81f40b2665f1c244086b967b3ff04ce36ccea8dc359d68c8  buildinfo
1.66.2 h1:yA++v1Kzm5mkDZWqNcGYDvKSzQza3pJCiUdatpatkq8= go1.22.7 amd64 linux
go	go1.22.7
path	fortio.org/fortio
mod	fortio.org/fortio	v1.66.2	h1:yA++v1Kzm5mkDZWqNcGYDvKSzQza3pJCiUdatpatkq8=
dep	fortio.org/cli	v1.9.0	h1:cPgNHvrjxznmbmwuXSwPqQLKZ+RMW8i0iAOESLjt1aI=
dep	fortio.org/dflag	v1.7.2	h1:lUhXFvDlw4CJj/q7hPv/TC+n/wVoQylzQO6bUg5GQa0=
dep	fortio.org/log	v1.16.0	h1:GhU8/9NkYZmEIzvTN/DTMedDAStLJraWUUVUA2EbNDc=
dep	fortio.org/scli	v1.15.2	h1:vWXt4QOViXNWy4Gdm7d2FDfptzWD00QiWzYAM/IUF7c=
dep	fortio.org/sets	v1.2.0	h1:FBfC7R2xrOJtkcioUbY6WqEzdujuBoZRbSdp1fYF4Kk=
dep	fortio.org/struct2env	v0.4.1	h1:rJludAMO5eBvpWplWEQNqoVDFZr4RWMQX7RUapgZyc0=
dep	fortio.org/version	v1.0.4	h1:FWUMpJ+hVTNc4RhvvOJzb0xesrlRmG/a+D6bjbQ4+5U=
dep	github.com/fsnotify/fsnotify	v1.7.0	h1:8JEhPFa5W2WU7YfeZzPNqzMP6Lwt7L2715Ggo0nosvA=
dep	github.com/golang/protobuf	v1.5.4	h1:i7eJL8qZTpSEXOPTxNKhASYpMn+8e5Q6AdndVa1dWek=
dep	github.com/google/uuid	v1.6.0	h1:NIvaJDMOsjHA8n1jAhLSgzrAzy1Hgr+hNrb57e+94F0=
dep	github.com/kortschak/goroutine	v1.1.2	h1:lhllcCuERxMIK5cYr8yohZZScL1na+JM5JYPRclWjck=
dep	golang.org/x/crypto/x509roots/fallback	v0.0.0-20240904212608-c9da6b9a4008	h1:vKHSxFhPLnBEYu9R8DcQ4gXq9EqU0VVhC9pq9wmtYsg=
dep	golang.org/x/exp	v0.0.0-20240904232852-e7e105dedf7e	h1:I88y4caeGeuDQxgdoFPUq097j7kNfw6uvuiNxUBfcBk=
dep	golang.org/x/net	v0.29.0	h1:5ORfpBpCs4HzDYoodCDBbwHzdR5UrLBZ3sOnUJmFoHo=
dep	golang.org/x/sys	v0.25.0	h1:r+8e+loiHxRqhXVl6ML1nO3l1+oFoWbnlu2Ehimmi34=
dep	golang.org/x/text	v0.18.0	h1:XvMDiNzPAl0jr17s6W9lcaIhGUfUORdGCNsuLmPG224=
dep	google.golang.org/genproto/googleapis/rpc	v0.0.0-20240903143218-8af14fe29dc1	h1:pPJltXNxVzT4pK9yD8vR9X75DaWYYmLGMsEvBfFQZzQ=
dep	google.golang.org/grpc	v1.66.0	h1:DibZuoBznOxbDQxRINckZcUvnCEvrW9pcWIE2yF9r1c=
dep	google.golang.org/protobuf	v1.34.2	h1:6xV6lTsCfpGD21XK49h7MhtcApnLqkfYgPcdHftf6hg=
build	-buildmode=exe
build	-compiler=gc
build	-ldflags=-s
build	DefaultGODEBUG=httplaxcontentlength=1,httpmuxgo121=1,tls10server=1,tlsrsakex=1,tlsunsafeekm=1
build	CGO_ENABLED=0
build	GOARCH=amd64
build	GOOS=linux
build	GOAMD64=v1
```
```
./fortio buildinfo
dev  go1.23.0 amd64 linux
go	go1.23.0
path	fortio.org/fortio
mod	fortio.org/fortio	(devel)	
dep	fortio.org/cli	v1.9.0	h1:cPgNHvrjxznmbmwuXSwPqQLKZ+RMW8i0iAOESLjt1aI=
dep	fortio.org/dflag	v1.7.2	h1:lUhXFvDlw4CJj/q7hPv/TC+n/wVoQylzQO6bUg5GQa0=
dep	fortio.org/log	v1.16.0	h1:GhU8/9NkYZmEIzvTN/DTMedDAStLJraWUUVUA2EbNDc=
dep	fortio.org/scli	v1.15.2	h1:vWXt4QOViXNWy4Gdm7d2FDfptzWD00QiWzYAM/IUF7c=
dep	fortio.org/sets	v1.2.0	h1:FBfC7R2xrOJtkcioUbY6WqEzdujuBoZRbSdp1fYF4Kk=
dep	fortio.org/struct2env	v0.4.1	h1:rJludAMO5eBvpWplWEQNqoVDFZr4RWMQX7RUapgZyc0=
dep	fortio.org/version	v1.0.4	h1:FWUMpJ+hVTNc4RhvvOJzb0xesrlRmG/a+D6bjbQ4+5U=
dep	github.com/fsnotify/fsnotify	v1.7.0	h1:8JEhPFa5W2WU7YfeZzPNqzMP6Lwt7L2715Ggo0nosvA=
dep	github.com/golang/protobuf	v1.5.4	h1:i7eJL8qZTpSEXOPTxNKhASYpMn+8e5Q6AdndVa1dWek=
dep	github.com/google/uuid	v1.6.0	h1:NIvaJDMOsjHA8n1jAhLSgzrAzy1Hgr+hNrb57e+94F0=
dep	github.com/kortschak/goroutine	v1.1.2	h1:lhllcCuERxMIK5cYr8yohZZScL1na+JM5JYPRclWjck=
dep	golang.org/x/crypto/x509roots/fallback	v0.0.0-20240904212608-c9da6b9a4008	h1:vKHSxFhPLnBEYu9R8DcQ4gXq9EqU0VVhC9pq9wmtYsg=
dep	golang.org/x/exp	v0.0.0-20240904232852-e7e105dedf7e	h1:I88y4caeGeuDQxgdoFPUq097j7kNfw6uvuiNxUBfcBk=
dep	golang.org/x/net	v0.29.0	h1:5ORfpBpCs4HzDYoodCDBbwHzdR5UrLBZ3sOnUJmFoHo=
dep	golang.org/x/sys	v0.25.0	h1:r+8e+loiHxRqhXVl6ML1nO3l1+oFoWbnlu2Ehimmi34=
dep	golang.org/x/text	v0.18.0	h1:XvMDiNzPAl0jr17s6W9lcaIhGUfUORdGCNsuLmPG224=
dep	google.golang.org/genproto/googleapis/rpc	v0.0.0-20240903143218-8af14fe29dc1	h1:pPJltXNxVzT4pK9yD8vR9X75DaWYYmLGMsEvBfFQZzQ=
dep	google.golang.org/grpc	v1.66.0	h1:DibZuoBznOxbDQxRINckZcUvnCEvrW9pcWIE2yF9r1c=
dep	google.golang.org/protobuf	v1.34.2	h1:6xV6lTsCfpGD21XK49h7MhtcApnLqkfYgPcdHftf6hg=
build	-buildmode=exe
build	-compiler=gc
build	DefaultGODEBUG=asynctimerchan=1,gotypesalias=0,httplaxcontentlength=1,httpmuxgo121=1,httpservecontentkeepheaders=1,tls10server=1,tls3des=1,tlskyber=0,tlsrsakex=1,tlsunsafeekm=1,winreadlinkvolume=0,winsymlink=0,x509keypairleaf=0,x509negativeserial=1
build	CGO_ENABLED=1
build	CGO_CFLAGS=
build	CGO_CPPFLAGS=
build	CGO_CXXFLAGS=
build	CGO_LDFLAGS=
build	GOARCH=amd64
build	GOOS=linux
build	GOAMD64=v1
build	vcs=git
build	vcs.revision=582d82a75a3172cc40eb0fe8e807c753b2a53770
build	vcs.time=2024-09-06T16:50:50Z
build	vcs.modified=true
```
The error is the same with both the container image and the self compiled binary.
When using the `-stdclient` option it seams that i cannot reproduce the problem (neither with the container image nor with the self compiled binary).

The debug logs for my self compiled binary are the following:
```
./fortio load -n 1 -c 1 --qps 0 -keepalive=false -loglevel debug https://somehost/api/infos
08:17:51.687 r1 [INF] logger.go:290> Log level is now 0 Debug (was 2 Info)
08:17:51.687 r1 [INF] scli.go:122> Starting, command="Φορτίο", version="dev  go1.23.0 amd64 linux", go-max-procs=8
08:17:51.688 r1 [VRB] stats.go:538> Will use [50 75 90 99 99.9] for percentiles
08:17:51.688 r1 [VRB] commonflags.go:183> Will verify TLS certificates, use -k / -https-insecure to disable
Fortio dev running at 0 queries per second, 8->8 procs, for 1 calls: https://somehost/api/infos
08:17:51.688 r1 [VRB] fortio_main.go:418> Generated Labels: somehost/api/infos , user
08:17:51.688 r1 [INF] httprunner.go:121> Starting http test, run=0, url="https://somehost/api/infos", threads=1, qps="-1.0", warmup="parallel", conn-reuse=""
08:17:51.688 r1 [VRB] periodic.go:308> Negative qps -1.000000 means max speed mode/no wait between calls
08:17:51.688 r1 [VRB] periodic.go:182> NewAborter called 0xc0002ca180 {Aborter 0xc0002ca180 stopChan 0xc0001123f0 startChan 0xc000112460 hasStarted false stopRequested false}
08:17:51.688 r1 [VRB] http_client.go:148> URLSchemeCheck &{TLSOptions:{Insecure:false MTLS:false CACert: Cert: Key: UnixDomainSocket:} URL:https://somehost/api/infos NumConnections:1 Compression:false DisableFastClient:false HTTP10:false H2:false DisableKeepAlive:true AllowHalfClose:false FollowRedirects:false initDone:true https:false Resolve: extraHeaders:map[] hostOverride: HTTPReqTimeOut:3s UserCredentials: ContentType: Payload:[] MethodOverride: LogErrors:true ID:0 UniqueID:0 SequentialWarmup:false ConnReuseRange:[0 0] NoResolveEachConn:false Offset:0s Resolution:0.001 ClientTrace:<nil> Transport:<nil> PayloadReader:<nil> DataWriter:<nil>}
08:17:51.688 r1 [DBG] http_client.go:788> NewFastClient GET https://somehost/api/infos
08:17:51.688 r8 [VRB] periodic.go:347> WATCHER 1 First outstanding run starting, catching signal
08:17:51.688 r1 [VRB] http_client.go:849> [0] No port specified, using https
08:17:51.688 r1 [DBG] network.go:354> Resolve() called with host=somehost port=https proto=tcp
08:17:51.688 r8 [VRB] periodic.go:353> WATCHER 1 starting new watcher for signal! chan  g 0xc0001d0000 r 0xc0001123f0 (3)
08:17:51.689 r1 [VRB] network.go:405> Resolved somehost:https to tcp ip4 cached-rr #0 addr 0.0.0.0:443
08:17:51.689 r1 [DBG] http_client.go:908> [0] Created client:
0.0.0.0:443
GET /api/infos HTTP/1.1
Host: somehost
Connection: close
User-Agent: fortio.org/fortio-dev


08:17:51.689 r1 [VRB] periodic.go:156> RUNNER starting... can now be Abort()ed, telling {Aborter 0xc0002ca180 stopChan 0xc0001123f0 startChan 0xc000112460 hasStarted true stopRequested false} - 0xc000112460
Starting at max qps with 1 thread(s) [gomax 8] for exactly 1 calls (1 per thread + 0)
08:17:51.689 r1 [VRB] periodic.go:547> Running single threaded
08:17:51.689 r1 [DBG] httprunner.go:63> Calling in 0
08:17:51.696 r1 [DBG] stats.go:293> v 0.007068 -> scaledVal 7.06767900000000004 ceil 8.000000 delta 0.06767900000000004 - svInt 7
08:17:51.696 r1 [VRB] network.go:485> Not setting socket options on non tcp socket 0.0.0.0:443
08:17:51.702 r1 [ERR] http_client.go:1098> Read error, err="EOF", size=0, header_len=0, dest={"IP":"0.0.0.0","Port":443,"Zone":""}, url="https://somehost/api/infos", thread=0, run=0
08:17:51.702 r1 [DBG] http_client.go:1268> [0] Closed ok from 0.0.0.0:443 after reading 0 bytes
08:17:51.702 r1 [DBG] httprunner.go:65> Got in  -1 hsz 0 sz 0 - will abort on 0
08:17:51.702 r1 [DBG] stats.go:293> v 0.012523 -> scaledVal 12.52286099999999891 ceil 13.000000 delta 0.5228609999999989 - svInt 12
08:17:51.702 r1 [DBG] stats.go:293> v 0.012523 -> scaledVal 12.52286099999999891 ceil 13.000000 delta 0.5228609999999989 - svInt 12
08:17:51.702 r1 [INF] periodic.go:851> T000 ended after 12.563608ms : 1 calls. qps=79.59496985260921
08:17:51.702 r1 [INF] stats.go:443> T000 Function duration : count 1 avg 0.012522861 +/- 0 min 0.012522861 max 0.012522861 sum 0.012522861
# range, mid point, percentile, count
>= 0.0125229 <= 0.0125229 , 0.0125229 , 100.00, 1
# target 99% 0.0125229

08:17:51.702 r1 [INF] stats.go:443> T000 Sleep time : no data

Ended after 12.597082ms : 1 calls. qps=79.383
08:17:51.702 r1 [INF] periodic.go:581> Run ended, run=0, elapsed=12597082, calls=1, qps=79.38346356719754
Aggregated Function Time : count 1 avg 0.012522861 +/- 0 min 0.012522861 max 0.012522861 sum 0.012522861
# range, mid point, percentile, count
>= 0.0125229 <= 0.0125229 , 0.0125229 , 100.00, 1
# target 50% 0.0125229
# target 75% 0.0125229
# target 90% 0.0125229
# target 99% 0.0125229
# target 99.9% 0.0125229
Error cases : count 1 avg 0.012522861 +/- 0 min 0.012522861 max 0.012522861 sum 0.012522861
# range, mid point, percentile, count
>= 0.0125229 <= 0.0125229 , 0.0125229 , 100.00, 1
# target 50% 0.0125229
# target 75% 0.0125229
# target 90% 0.0125229
# target 99% 0.0125229
# target 99.9% 0.0125229
08:17:51.702 r1 [VRB] periodic.go:625> RUNNER aborter not already closed, closing
08:17:51.702 r1 [VRB] periodic.go:389> Abort called for 0xc0003002a0
08:17:51.702 r1 [VRB] periodic.go:115> ABORT Closing already started or not waiting {Aborter 0xc0002ca180 stopChan 0xc0001123f0 startChan 0xc000112460 hasStarted true stopRequested true}
08:17:51.702 r1 [VRB] periodic.go:120> ABORT reading start channel
08:17:51.702 r1 [VRB] periodic.go:124> ABORT done reading start channel, got true
08:17:51.702 r1 [VRB] periodic.go:171> RUNNER reset: we were Abort()ed already, start chan empty
# Socket and IP used for each connection:
08:17:51.702 r1 [DBG] http_client.go:773> [0] Closing 0xc000190c60 https://somehost/api/infos socket count 1
[0]   1 socket used, resolved to 0.0.0.0:443, connection timing : count 1 avg 0.007067679 +/- 0 min 0.007067679 max 0.007067679 sum 0.007067679
Connection time histogram (s) : count 1 avg 0.007067679 +/- 0 min 0.007067679 max 0.007067679 sum 0.007067679
# range, mid point, percentile, count
>= 0.00706768 <= 0.00706768 , 0.00706768 , 100.00, 1
# target 50% 0.00706768
# target 75% 0.00706768
# target 90% 0.00706768
# target 99% 0.00706768
# target 99.9% 0.00706768
08:17:51.702 r8 [VRB] periodic.go:368> WATCHER 1 r.Stop readable
08:17:51.702 r8 [VRB] periodic.go:371> WATCHER 1 End of go routine
08:17:51.702 r8 [VRB] periodic.go:375> WATCHER 1 Last watcher: resetting signal handler
Sockets used: 1 (for perfect keepalive, would be 1)
Uniform: false, Jitter: false, Catchup allowed: true
IP addresses distribution:
0.0.0.0:443: 1
Code  -1 : 1 (100.0 %)
Response Header Sizes Histogram : count 1 avg 0 +/- 0 min 0 max 0 sum 0
# range, mid point, percentile, count
>= 0 <= 0 , 0 , 100.00, 1
Response Body/Total Sizes Histogram : count 1 avg 0 +/- 0 min 0 max 0 sum 0
# range, mid point, percentile, count
>= 0 <= 0 , 0 , 100.00, 1
08:17:51.702 r1 [VRB] periodic.go:389> Abort called for 0xc0003002a0
08:17:51.702 r1 [VRB] periodic.go:108> ABORT already aborted {Aborter 0xc0002ca180 stopChan <nil> startChan 0xc000112460 hasStarted false stopRequested false}
All done 1 calls (plus 0 warmup) 12.523 ms avg, 79.4 qps
```

As far as I can see, #974 fixes the problem as I was no longer able to reproduce it.

**Comment by ldemailly** [comment]
Thanks a lot! will ship that fix shortly then.

**Comment by ldemailly** [comment]
fixed in 1.66.3, and thanks again for having reported the issue

**Comment by hellivan** [comment]
@ldemailly Many thanks for your help and commitment.
